### PR TITLE
.github/workflows: add codespell workflow

### LINKED ISSUE #320: Add Codespell GitHub action
To avoid further common typos being added to the codebase, pipeline like https://github.com/rook/rook/pull/6564 could be added to automatically check for most common typos.

### PR DESCRIPTION
**Pull request by invidian**
#### What this PR does / why we need it:

To avoid further common typos being added to the codebase.

#### Which issue this PR fixes
- Closes #320

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

#### Special notes for your reviewer:

Right now it is expected that this pipeline will fail until PRs #315 #316 #317 #318 #319 are merged and this PR is rebased.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by invidian** [comment]
Spellchecking is passing now :tada:

**Review by monotek — APPROVED** [review]

**Comment by invidian** [comment]
Can this be merged then?

File: .github/workflows/codespell.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+jobs:
+  codespell:
+    name: Codespell
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Codespell
+        uses: codespell-project/actions-codespell@master
```
Comment (torstenwalter) [inline .github/workflows/codespell.yaml:16]: Should we pin it to a dedicated version?

File: .github/workflows/codespell.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        with:
+          skip: .git,./charts/kube-prometheus-stack/crds/crd-prometheuses.yaml,./charts/kube-prometheus-stack/crds/crd-alertmanagers.yaml,./charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/apiserver.yaml
```
Comment (torstenwalter) [inline .github/workflows/codespell.yaml:18]: Why are we skipping those? Are there spelling mistakes?

Comment (invidian) [reply .github/workflows/codespell.yaml:18]: Yes, there are. Those files are coming from upstream, so they shouldn't be modified here. I already fixed those typos upstream (prometheus-operator/prometheus-operator#3653), but I don't know how to pull those changes in here, as sync script is currently broken (#225).

File: .github/workflows/codespell.yaml (hunk 1 above)
Comment (invidian) [reply .github/workflows/codespell.yaml:16]: As far as I see upstream has no releases. Would you like me to pin it to specific commit?

**Review by torstenwalter — APPROVED** [review]
lgtm - I think we can merge it as is. The check is currently not marked as mandatory.
So we will see the result as part of PR status checks, but it's still possible to bypass it.

We could leave it like this for a while to see how it goes and then decide to make it mandatory.

**Review by torstenwalter — APPROVED** [review]

**Pull request merged**